codersdk/oauth2.go

```diff
@@ -168,3 +179,50 @@ func (c *Client) DeleteOAuth2ProviderAppSecret(ctx context.Context, appID uuid.U
 	}
 	return nil
 }
+
+type OAuth2ProviderGrantType string
+
+const (
+	OAuth2ProviderGrantTypeAuthorizationCode OAuth2ProviderGrantType = "authorization_code"
+	OAuth2ProviderGrantTypeRefreshToken      OAuth2ProviderGrantType = "refresh_token"
+)
+
+func (e OAuth2ProviderGrantType) Valid() bool {
+	switch e {
+	case OAuth2ProviderGrantTypeAuthorizationCode, OAuth2ProviderGrantTypeRefreshToken:
+		return true
+	}
+	return false
+}
+
+type OAuth2ProviderResponseType string
+
+const (
+	OAuth2ProviderResponseTypeCode OAuth2ProviderResponseType = "code"
+)
+
+func (e OAuth2ProviderResponseType) Valid() bool {
+	//nolint:gocritic,revive // More cases might be added later.
+	switch e {
+	case OAuth2ProviderResponseTypeCode:
+		return true
+	}
+	return false
+}
+
+// RevokeOAuth2ProviderApp completely revokes an app's access for the user.
+func (c *Client) RevokeOAuth2ProviderApp(ctx context.Context, appID uuid.UUID) error {
+	res, err := c.Request(ctx, http.MethodDelete, "/login/oauth2/tokens", nil, func(r *http.Request) {
```

Comment:
Oh really good point, let me change it now real quick.

Reply:
Oh right I had it here to have the post/delete paradigm on the same endpoint.  Maybe I should actually move this back to `/api/v2` since it is our own thing, not part of the whole oauth flow.

Reply:
This is not an oauth2 spec endpoint right? 

Eh just keep it here and comment that it is not oauth 2 spec. If you feel this has to be a static endpoint though, then just comment why it should be. I'll defer to your judgement